client/Components/GuestDropdown.jsx

```diff
@@ -11,15 +19,18 @@ class GuestDropdown extends React.Component {
       infants: 0,
       expand: false, // false in production
     };
+    this.buttonHandler = this.buttonHandler.bind(this);
     this.expanded = this.expanded.bind(this);
   }
 
-  clickHander(e) {
-
+  buttonHandler(e) {
+    e.persist();
+    const expand = !this.state.expand;
+    this.setState({ expand });
```

Comment:
https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous